### PR TITLE
Add implications of domain model

### PR DESCRIPTION
**Pull request by lesser1**

### CODE REVIEW

**Review by nk-coding — CHANGES_REQUESTED** [review]
prettier is unhappy

**Review by nk-coding — APPROVED** [review]

File: docs/dev-manuals/architecture/domain-model.mdx [hunk 1]
```diff
@@ -13,3 +13,50 @@ import CustomImage from "@site/src/components/CustomImage";
...
+- If an `Invoice` has an associated `Payment`, then the payment has been made -- the store has received the money. If not, then the payment is still due. This implies that the association between `Invoice` and `Payment` fully models the payment's status.
+- The store exclusively supports payment methods which result in the payment being processed before the shipping of any ordered items. This implies that the user must always pay first before they can receive the order.
+
+### Regarding _Inventory_
+
+Since the domain model does not account for returned items, this implies that returned items will simply be thrown away after they have been returned to the _Inventory_.
+
+### Regarding _Category_
+
+The store has only one level of categories but a product can be part of multiple categories.
```
Comment (delvh) [inline docs/dev-manuals/architecture/domain-model.mdx]: I have no idea what you're trying to say here.

Comment (delvh) [reply docs/dev-manuals/architecture/domain-model.mdx]: Also, that is not necessarily true

Comment (legendofa) [reply docs/dev-manuals/architecture/domain-model.mdx]: Categories are not hierarchical, so there are no subcategories

Comment (delvh) [reply docs/dev-manuals/architecture/domain-model.mdx]:
```suggestion
The store has no category inheritance.
All categories are separate.
```

**Pull request merged**